### PR TITLE
rfc: errors schema

### PR DESCRIPTION
**Pull request by iamdey**
May be cornice will us better than I propose. What do you think @gdchamal ?

### CODE REVIEW

**Comment by gdchamal** [comment]
Yes we must define error format as produced by colander, the validation framework used by cornice API

But I'm must wrap validation error on business logic layer (caliopen.base.*) to be formatted this way.

Also we must include HTTP status code in this RFC, this is basically a BAD REQUEST (400) for such schema validation error.

**Comment by iamdey** [comment]
ok it sounds colander can do this: http://docs.pylonsproject.org/projects/colander/en/latest/basics.html#deserializing-an-invalid-serialization

but it requires to be l10n (didn't saw anything about that): 

```
{'age':'-1 is less than minimum value 0',
 'friends.1.0':'"t" is not a number',
 'phones.0.location:'"bar" is not one of "home", "work"'}
```

File: _posts/rfc/2016-04-18-errors-schema.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+### Error
+
+* An Error MUST have a description in English (eg. `The login is not available`)
+* An Error MUST be typed (the kind of error, eg. `max-len`, `internal`)
+* An Error MAY have one or more values depends on type (eg. for error `max`, value: `6`)
+
+#### Property
+
+In a modifying request (POST, PUT or PATCH), an Error MAY have a `property` that represents an invalid resource property.
```
Comment (iamdey) [inline _posts/rfc/2016-04-18-errors-schema.md]: you right the status code is missing here

**Comment by gdchamal** [comment]
:+1:

**Pull request merged**